src/main/java/us/ihmc/mecano/algorithms/JointTorqueRegressorCalculator.java

```diff
@@ -191,11 +189,125 @@ public void compute()
       collectJointTorqueRegressorFromRecursionSteps();
    }
 
+   /**
+    * Computes the n-by-10 block of the joint torque regressor matrix corresponding to {@code body} for the current state of the multi-body system
+    * {@code input}, where n is the number of DoFs.
+    * <p>
+    * The position, velocity, and acceleration of the joints are extracted directly from the joint state.
+    * </p>
+    *
+    * @param body the rigid body to compute the joint torque regressor matrix block for.
+    */
+   public void compute(RigidBodyReadOnly body)
+   {
+      // Set all spatial inertias to zero
+      initialRecursionStep.setSpatialInertiasToZeroRecursively();
+
+      // Perform an initial forward pass of the inverse dynamics calculator (which roughly corresponds to the forward
+      // kinematics)
+      inverseDynamicsCalculator.initializeJointAccelerationMatrix(null);
+      inverseDynamicsCalculator.initialRecursionStep.passOneRecursive();
+
+      rigidBodyToRecursionStepMap.get(body).markUpstreamAsModifiedRecursively();
```

Comment:
Keep in mind that retrieving a value in a map is not as trivial as accessing the i-th element of an array. If you need to interact multiple times with the same value, I'd suggest storing it in a local variable so you perform only 1 lookup.
Out of curiosity, just take a peek at the algorithm to retrieve a value in a `HashMap`: https://github.com/openjdk/jdk/blob/master/src/java.base/share/classes/java/util/HashMap.java#L565

Reply:
That's awesome stuff, done.